astroid/interpreter/_import/spec.py

```diff
@@ -121,6 +121,8 @@ class ExplicitNamespacePackageFinder(ImpFinder):
     """A finder for the explicit namespace packages, generated through pkg_resources."""
 
     def find_module(self, modname, module_parts, processed, submodule_path):
+        if submodule_path:
```

Comment:
I'm not sure this is the problem or at least, I can't demonstrate it so far. From what I'm noticing, ``submodule_path`` contains the filename provided by the ``ImpFinder``, which doesn't make sense, since a file is not a valid search location. Secondly, it seems that in this test case, the submodule_path provided is the current directory. Instead of just returning when ``submodule_path`` is not empty, we could check if the submodule_path passed in this function points to different root directories than the submodule path found by looking in the module itself. Do you think this makes sense? can you check it?